File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
        const maxCandies = candies.reduce((ele, curMax) => Math.max(ele, curMax), 0);
    
        const res = [];
    
        for(let candiesCount of candies){
            res.push(candiesCount + extraCandies >= maxCandies);
        }
        
        return res;
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maxCandies = Arrays.stream(candies).max().getAsInt();
        List<Boolean> res = new ArrayList<>();
        
        for(int candiesCount : candies){
            if(candiesCount + extraCandies >= maxCandies){
                res.add(true);
            }
            else{
                res.add(false);
            }
        }
        
        return res;
    }
}